File: CosmeticaShop.Web/Areas/Admin/assets/js/Form/Order/OrderProductModel.js
var Order = Order || {};

(function () {

    if (Order.OrderProductModel) {
        console.error("Order.OrderProductModel уже был создан");
        return;
    }

    /**
     * Класс для описания модели шапки заказа
     * @param {     
     * } theParams 
     * @returns {} 
     */
    Order.OrderProductModel = function (theParams) {
        theParams = theParams || {};

        this.Id = ko.observable(theParams.Id || "");
        this.ProductId = ko.observable(theParams.ProductId || "");
        this.ProductName = ko.observable(theParams.ProductName || "");
        this.PhotoUrl = ko.observable(theParams.PhotoUrl || "");
        this.Quantity = ko.observable(theParams.Quantity || 0);
        this.Price = ko.observable(theParams.Price || 0);
        this.Discount = ko.observable(theParams.Discount || 0);
        this.Amount = ko.observable(theParams.Amount || 0);

        return this;
    };

    /**
     * Определяем конструктор
     */
    Order.OrderProductModel.prototype.constructor = Order.OrderProductModel;

    Order.OrderProductModel.prototype.log = function (text) {
        console.log("Ошибка в классе Order.OrderProductModel: " + text);
    };

    Order.OrderProductModel.prototype.GetData = function () {
        var model = {
            Id: this.Id(),
            ProductId: this.ProductId(),
            Quantity: this.Quantity(),
            Price: this.Price(),
            Discount: this.Discount(),
            Amount: this.Amount(),
        };
    }
})();

File: CosmeticaShop.IServices/Models/Order/Admin/OrderProductModel.cs
using System;
using CosmeticaShop.IServices.Models.Product;

namespace CosmeticaShop.IServices.Models.Order.Admin
{
    public class OrderProductModel
    {
        /// <summary>
        /// Ид
        /// </summary>
        public int Id { get; set; }
        /// <summary>
        /// Ид товара
        /// </summary>
        public int ProductId { get; set; }
        /// <summary>
        /// Наименование товара
        /// </summary>
        public string ProductName { get; set; }
        /// <summary>
        /// Url изображения
        /// </summary>
        public string PhotoUrl { get; set; }
        /// <summary>
        /// Количество
        /// </summary>
        public int Quantity { get; set; }
        /// <summary>
        /// Цена
        /// </summary>
        public decimal Price { get; set; }
        /// <summary>
        /// Скидка на товар
        /// </summary>
        public decimal Discount { get; set; }
        /// <summary>
        /// Сумма для платежа
        /// </summary>
        public decimal Amount { get; set; }
        /// <summary>
        /// Цена со скидкой
        /// </summary>
        public decimal DiscountPrice { get; set; }
    }
}
